File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Order.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass
import java.util.UUID

@JsonClass(generateAdapter = true)
data class Order(
    val id: Int,
    val order_number: String,
    val employee_id: Int? = null,
    val employee_name: String? = null,
    val status: String,
    val subtotal: Double? = null,
    val tax: Double? = null,
    val tip: Double? = null,
    val total: Double? = null,
    val payment_intent_id: String? = null,
    val payment_status: String? = null,
    val payment_method: String? = null,
    val source: String? = null,
    val created_at: String? = null,
    val completed_at: String? = null,
    val items: List<OrderItem>? = null
)

@JsonClass(generateAdapter = true)
data class OrderItem(
    val id: Int? = null,
    val order_id: Int? = null,
    val menu_item_id: Int? = null,
    val item_name: String,
    val quantity: Int,
    val unit_price: Double? = null,
    val notes: String? = null,
    val combo_instance_id: String? = null,
    val modifiers: List<OrderItemModifier>? = null
)

@JsonClass(generateAdapter = true)
data class OrderItemModifier(
    val id: Int? = null,
    val order_item_id: Int? = null,
    val modifier_id: Int? = null,
    val modifier_name: String,
    val price_adjustment: Double = 0.0
)

data class CartItem(
    val cartId: String = UUID.randomUUID().toString(),
    val menuItemId: Int,
    val itemName: String,
    var quantity: Int,
    val unitPrice: Double,
    var notes: String? = null,
    val comboInstanceId: String? = null,
    val menuItem: MenuItem? = null,
    val selectedModifierIds: List<Int>? = null,
    val selectedModifierNames: List<String>? = null
) {
    val lineTotal: Double get() = unitPrice * quantity
}

@JsonClass(generateAdapter = true)
data class CreateOrderRequest(
    val employee_id: Int,
    val items: List<CreateOrderItem>
)

@JsonClass(generateAdapter = true)
data class CreateOrderItem(
    val menu_item_id: Int,
    val quantity: Int,
    val notes: String? = null,
    val modifiers: List<Int>? = null,
    val combo_instance_id: String? = null
)

File: apps/pos/native-windows/DesktopKitchenPOS/Models/Order.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record Order
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("order_number")]
    public string OrderNumber { get; init; } = "";

    [JsonPropertyName("employee_id")]
    public int? EmployeeId { get; init; }

    [JsonPropertyName("employee_name")]
    public string? EmployeeName { get; init; }

    [JsonPropertyName("status")]
    public string Status { get; init; } = "pending";

    [JsonPropertyName("subtotal")]
    public double? Subtotal { get; init; }

    [JsonPropertyName("tax")]
    public double? Tax { get; init; }

    [JsonPropertyName("tip")]
    public double? Tip { get; init; }

    [JsonPropertyName("total")]
    public double? Total { get; init; }

    [JsonPropertyName("payment_intent_id")]
    public string? PaymentIntentId { get; init; }

    [JsonPropertyName("payment_status")]
    public string? PaymentStatus { get; init; }

    [JsonPropertyName("payment_method")]
    public string? PaymentMethod { get; init; }

    [JsonPropertyName("source")]
    public string? Source { get; init; }

    [JsonPropertyName("created_at")]
    public string? CreatedAt { get; init; }

    [JsonPropertyName("completed_at")]
    public string? CompletedAt { get; init; }

    [JsonPropertyName("items")]
    public List<OrderItem>? Items { get; init; }
}

public record OrderItem
{
    [JsonPropertyName("id")]
    public int? Id { get; init; }

    [JsonPropertyName("order_id")]
    public int? OrderId { get; init; }

    [JsonPropertyName("menu_item_id")]
    public int? MenuItemId { get; init; }

    [JsonPropertyName("item_name")]
    public string ItemName { get; init; } = "";

    [JsonPropertyName("quantity")]
    public int Quantity { get; init; }

    [JsonPropertyName("unit_price")]
    public double? UnitPrice { get; init; }

    [JsonPropertyName("notes")]
    public string? Notes { get; init; }

    [JsonPropertyName("combo_instance_id")]
    public string? ComboInstanceId { get; init; }

    [JsonPropertyName("modifiers")]
    public List<OrderItemModifier>? Modifiers { get; init; }
}

public record OrderItemModifier
{
    [JsonPropertyName("id")]
    public int? Id { get; init; }

    [JsonPropertyName("order_item_id")]
    public int? OrderItemId { get; init; }

    [JsonPropertyName("modifier_id")]
    public int? ModifierId { get; init; }

    [JsonPropertyName("modifier_name")]
    public string ModifierName { get; init; } = "";

    [JsonPropertyName("price_adjustment")]
    public double PriceAdjustment { get; init; }
}

public class CartItem
{
    public string CartId { get; set; } = Guid.NewGuid().ToString();
    public int MenuItemId { get; set; }
    public string ItemName { get; set; } = "";
    public int Quantity { get; set; }
    public double UnitPrice { get; set; }
    public string? Notes { get; set; }
    public MenuItem? MenuItemRef { get; set; }
    public List<int>? SelectedModifierIds { get; set; }
    public List<string>? SelectedModifierNames { get; set; }

    public double LineTotal => UnitPrice * Quantity;
}

public record CreateOrderRequest
{
    [JsonPropertyName("employee_id")]
    public int EmployeeId { get; init; }

    [JsonPropertyName("items")]
    public List<CreateOrderItem> Items { get; init; } = new();
}

public record CreateOrderItem
{
    [JsonPropertyName("menu_item_id")]
    public int MenuItemId { get; init; }

    [JsonPropertyName("quantity")]
    public int Quantity { get; init; }

    [JsonPropertyName("notes")]
    public string? Notes { get; init; }

    [JsonPropertyName("modifiers")]
    public List<int>? Modifiers { get; init; }

    [JsonPropertyName("combo_instance_id")]
    public string? ComboInstanceId { get; init; }
}
